Fix replacing markers

TFP-dev opened the pull request:
Based on #2794. Instead of adding ```engineReinitMarkers```, it executes 3D markers reinit function when replacing or restoring models with id 1316 (ring), 1317 (cylinder and checkpoint) and 1559 (arrow).


![markers](https://github.com/multitheftauto/mtasa-blue/assets/72511827/4fb63a82-db61-4671-8950-1cde31265aec)

### Test resource
[replace_markers_test_resource.zip](https://github.com/multitheftauto/mtasa-blue/files/11686873/replace_markers_test_resource.zip)

TFP-dev committed:
Add engineReinitMarkers()

TFP-dev committed:
Use "using" insted of "typedef" in C3DMarkersSA::ReinitMarkers()

TFP-dev committed:
Update "CLuaEngineDefs::EngineReinitMarkers"

patrikjuvonen committed:
Merge branch 'master' into engineReinitMarkers

TFP-dev committed:
Merge branch 'master' into engineReinitMarkers

TFP-dev committed:
Merge branch 'master' into engineReinitMarkers

TFP-dev committed:
Remove engineReinitMarkers

TFP-dev committed:
Add ReinitMarkers to model replacing and restoring

TheNormalnij commented on Client/mods/deathmatch/logic/CClientDFF.cpp:
```diff
@@ -280,6 +284,10 @@ void CClientDFF::InternalRestoreModel(unsigned short usModel)
+
+    // 'Restream' 3D markers
+    if (usModel == 1316 || usModel == 1317 || usModel == 1559)
+        g_pClientGame->ReinitMarkers();
```
Can be better DI here:
```c++
else if (CClientMarkerManager::IsMarkerModel(usModel)) {
    m_pManager->GetMarkerManager()->ReinitMarkers();
}
```

TFP-dev replied:
Done

TheNormalnij commented on Client/mods/deathmatch/logic/CClientDFF.cpp:
```diff
@@ -109,6 +109,10 @@ bool CClientDFF::ReplaceModel(unsigned short usModel, bool bAlphaTransparency)
+    // 'Restream' 3D markers
+    if (usModel == 1316 || usModel == 1317 || usModel == 1559)
+        g_pClientGame->ReinitMarkers();
+
```
Should be in `DoReplaceModel` method with better DI.

TFP-dev replied:
Done

patrikjuvonen committed:
Merge branch 'master' into fix-replacing-markers

TFP-dev committed:
Merge branch 'master' into fix-replacing-markers

TFP-dev committed:
Move marker model checking into separate function

botder commented on Client/mods/deathmatch/logic/CClientDFF.cpp:
```diff
@@ -164,6 +164,15 @@ bool CClientDFF::DoReplaceModel(unsigned short usModel, bool bAlphaTransparency)
+            bool bResult = ReplaceObjectModel(pClump, usModel, bAlphaTransparency);
+
+            // 'Restream' 3D markers
+            g_pClientGame->ReinitMarkers();
```
Only do this when `bResult` is true.

botder commented on Client/mods/deathmatch/logic/CClientDFF.cpp:
```diff
@@ -164,6 +164,15 @@ bool CClientDFF::DoReplaceModel(unsigned short usModel, bool bAlphaTransparency)
         }
+        else if (CClientMarkerManager::IsMarkerModel(usModel))
+        {
+            bool bResult = ReplaceObjectModel(pClump, usModel, bAlphaTransparency);
```
Rename `bResult` to `wasReplaced`.

TFP-dev committed:
Merge branch 'master' into fix-replacing-markers

TFP-dev committed:
Rename 'bResult' to 'wasReplaced'

Rename 'bResult' to 'wasReplaced' and reinit markers only after successful model replacement.
